tests/integration/collective/collective/collective.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from enum import IntEnum
+from typing import Any, Tuple
+
+import legate.core.types as ty
+from legate.core import Rect, ReductionOp, Store
+
+from .library import user_context as context, user_lib  # type: ignore
+
+
+class OpCode(IntEnum):
+    COLLECTIVE = user_lib.cffi.COLLECTIVE
+
+
+def create_int64_store(shape: Tuple[Any, ...]) -> Store:
+    store = context.create_store(ty.int64, shape=shape)
+    # call empty function on the store to make Legion to think
+    # that we initialized the store
+    task = context.create_auto_task(
+        OpCode.COLLECTIVE,
+    )
+    task.add_output(store)
+    task.execute()
+    return store
+
+
+def _broadcast(store: Store, shape: Tuple[Any, ...]) -> Store:
```

Comment:
why do we need this general broadcast implementation here? I feel we could just directly transform the stores below.